lib/swaps/Swaps.ts

```diff
@@ -150,6 +165,40 @@ class Swaps extends EventEmitter {
     return;
   }
 
+  /**
+   * Saves deal to database and deletes from memory.
+   * @param deal The deal to persist.
+   */
+  private persistDeal = (deal: SwapDeal) => {
+    const { r_hash } = deal;
+    if (this.hashs.has(r_hash)) {
+      const persistDeal: SwapDealAttributes = {
```

Comment:
Do we need this separate object? I think you can just pass the `deal` to the `addSwapDeal` method.